raiden/network/pathfinding.py

```diff
@@ -167,11 +167,25 @@ def get_random_pfs(
     Returns a tuple of the chosen services url and eth address.
     If there are no PFS in the given registry, it returns (None, None).
     """
-    count = service_registry.service_count(block_identifier=block_identifier)
-    if count == 0:
+    count = service_registry.ever_made_deposits_len(block_identifier=block_identifier)
+    addresses = (
+        service_registry.ever_made_deposits(block_identifier=block_identifier, index=idx)
+        for idx in range(0, count)
+    )
+    address_list = list(
```

Comment:
This can be greatly simplified with a list comprehension:
```python
address_list = [
  addr
  for addr in addresses
  if service_registry.has_valid_registration(address=addr, block_identifier=block_identifier)
]
```

Reply:
OK.  I take that idea.

Reply:
Actually, the above can be even simpler with:

```python
addresses = []
for index in range(count):
    address = service_registry.ever_made_deposits(
        block_identifier=block_identifier,
        index=index,
    )
    is_valid = service_registry.has_valid_registration(
        address=address,
        block_identifier=block_identifier,
    )
    if is_valid:
        addresses.append(address)
```

If we could have ordered addresses it would be even simpler.

Reply:
Okay, third proposal, we can switch it around and reduce the number of RPC calls:

```python
number_of_addresses = service_registry.ever_made_deposits_len(
    block_identifier=block_identifier,
)
addresses_to_try = list(range(number_of_addresses))
random.shuffle(addresses_to_try)

while addresses_to_try:
    index = addresses_to_try.pop()
    address = service_registry.ever_made_deposits(
        block_identifier=block_identifier,
        index=index,
    )
    is_valid = service_registry.has_valid_registration(
        address=address,
        block_identifier=block_identifier,
    )
    if is_valid:
        return address
return None
```

Reply:
I'll do this with `indices_to_try`.

Reply:
I followed this somehow.